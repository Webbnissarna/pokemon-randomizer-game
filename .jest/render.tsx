import { render as rtlRender } from "@testing-library/react";
import { InitializeColorMode, ThemeProvider } from "theme-ui";

import theme from "../utils/theme";

import "./style.css";

export function render(ui: React.ReactElement) {
  return rtlRender(
    <>
      <InitializeColorMode />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  );
}
