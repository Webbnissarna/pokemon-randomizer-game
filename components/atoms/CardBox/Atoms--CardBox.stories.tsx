import { Heading, Text } from "theme-ui";
import CardBox from ".";

export function Default() {
  return (
    <CardBox>
      <Heading>Card Box</Heading>
      <Text>Hello World</Text>
    </CardBox>
  );
}
