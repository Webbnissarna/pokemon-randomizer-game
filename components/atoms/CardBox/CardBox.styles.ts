import { ThemeUIStyleObject } from "theme-ui";

export const mainStyle: ThemeUIStyleObject = {
  flexDirection: "column",
  backgroundColor: "contentBackground",
  paddingX: "md",
  paddingY: "lg",
  borderRadius: "md",
  boxShadow: "md",
};
