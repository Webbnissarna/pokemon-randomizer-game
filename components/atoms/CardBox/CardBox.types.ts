export interface CardBoxProps {
  children: React.ReactNode;
}
