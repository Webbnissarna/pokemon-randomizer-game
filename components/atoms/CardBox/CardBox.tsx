import { Flex } from "theme-ui";
import { mainStyle } from "./CardBox.styles";
import type { CardBoxProps } from "./CardBox.types";

export default function CardBox({ children }: CardBoxProps): JSX.Element {
  return <Flex sx={mainStyle}>{children}</Flex>;
}
