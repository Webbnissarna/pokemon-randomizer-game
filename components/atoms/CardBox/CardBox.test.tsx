import { render } from "../../../.jest/render";
import { Text } from "theme-ui";
import CardBox from ".";

describe("CardBox", () => {
  it("renders", () => {
    const { asFragment } = render(
      <CardBox>
        <Text>Hello World</Text>
      </CardBox>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
