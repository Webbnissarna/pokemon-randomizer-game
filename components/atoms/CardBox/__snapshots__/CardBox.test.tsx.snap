// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBox renders 1`] = `
<DocumentFragment>
  <script>
    (function() { try {
  var mode = localStorage.getItem('theme-ui-color-mode');
  if (!mode) return
  document.documentElement.classList.add('theme-ui-' + mode);
} catch (e) {} })();
  </script>
  <div
    class="css-wimkxn-Box"
  >
    <span
      class="css-1peva9n-Box"
    >
      Hello World
    </span>
  </div>
</DocumentFragment>
`;
